README.md

```diff
@@ -2,15 +2,142 @@
 
 ## Learning Goals
 
--SWBAT 1
--SWBAT 2
+- Recognize the benefits of optional arguments
+- Define a method with optional arguments
+- Define a method with optional and required arguments
 
 ## Introduction
 
-## SWBAT 1
+If we revisit the coffee shop scenario, typically a name may be given when you
+place your order so that your name can be called when the order is ready. What
+happens when they don't ask for a name? Usually, they _default_ to calling out
+the order that was placed. This may not be as personalized to you, but it's
+better than not knowing when your order is ready!
+
+In Ruby, we now see that when a method accepts arguments, those arguments become
+required; an error is thrown when those arguments are not passed in on
+invocation of the method. So what do we do when we have optional information?
+
+When signing up on a website, you may notice that if there are personalization
+options like a profile picture. However, when one isn't provided, it defaults to
+a generic avatar image, like a silhouette of a person.
+
+!["Default Facebook Avatar"](https://i.stack.imgur.com/l60Hf.png)
+
+This can be done by using optional arguments with _default_ values. We're going
+to introduce how to create methods that take in optional arguments and cover why
+they're important to programming.
+
+## Recognize the Benefits of Optional Arguments
+
+We should constantly strive for our code to be dynamic and flexible. As
+programmers, we might be considered lazy, but why do more work than is
+necessary? We want the code we write to be re-usable, which is why we have
+principles such as DRY.
+
+If we define a method, `#greeting`, without arguments, it'll be very difficult
+to reuse this code in any other manner. Since that's way too much work for us,
+we'll define our method to take in an _argument_ of someone's name:
+
+```ruby
+def greeting(name)
+  puts "Hello, #{name}"
+end
+```
+
+But what if we don't know the name of the person we are trying to greet? We can
+make this method even more flexible by making the `name` argument optional. We
+do this by using optional, or default, arguments.
+
+## Define a Method With Optional Arguments
+
+Default arguments are easy to add, you simply assign them a default value with
```

Comment:
be careful w/ args versus parameters. beginners die here al the time.

Maybe "you simply assign _their corresponding parameters a value with `=` in the parameter list in the method signature_"

You get to use "method signature" bc you intro'd it in an earlier lesson (booya)